pkg/appgateserver/config/appgate_configs_reader.go

```diff
@@ -0,0 +1,66 @@
+package config
+
+import (
+	"net/url"
+
+	"gopkg.in/yaml.v2"
+)
+
+// YAMLAppGateConfig is the structure used to unmarshal the AppGateServer config file
+type YAMLAppGateConfig struct {
```

Comment:
```suggestion
// TODO_DOCUMENT(@red-0ne): Add proper README documentation for yaml config files.
type YAMLAppGateConfig struct {
```

Can you add these to others too?